src/sledo/optimiser.py

```diff
@@ -1,188 +1,130 @@
-        the new file to the data directory.
-    """
+    """The SLEDO optimiser class."""
 
     def __init__(
         self,
-        name,
-        simulation_class,
-        search_space,
-        data_dir=None,
-        input_file=None,
-    ):
-        """
-        Initialise class instance.
+        name: str,
+        evaluation_function: callable,
+        search_space: dict,
+        metric: str,
+        search_alg: type(tune.search.Searcher),
+        max_total_trials: int,
+        max_concurrent_trials: int = 1,
+        data_dir: str | pathlib.Path = None,
+    ) -> None:
+        """Initialise class instance.
 
-        Attributes
+        Parameters
         ----------
-        simulation_class : sledo.Simulation
-            The simulation class to use for the optimisation.
-        search_space : list[dict]
-            The search space for the optimisation.
-        data_dir : str | pathlib.Path
-            Path to the data directory for the optimisation.
-        input_file: str | pathlib.Path
-            Path to the simulation input file for the optimisation.
+        name (str):
+            The name of the instance.
+        evaluation_function (callable):
+            The function used to evaluate each design iteration. Must accept a
+            dict of parameters and their values for a given design and return
+            a dict of metrics for that design.
+        search_space (dict):
+            The search space for the optimisation, values must be set according
+            to the Ray Tune Search Space API.
+        metric (str):
+            The metric of interest, must match one of the keys used in the
+            output of the evaluation function.
+        search_alg (tune.search.Searcher):
+            The search algorithm to use, must be an instance of a subclass of
+            the Ray Tune Searcher base class.
+        max_total_trials(int):
+            The maximum number of total trials to run.
+        max_concurrent_trials (int, optional):
+            The maximum number of concurrent trials. Defaults to 1, in which
+            case the optimisation is entirely sequential.
+        data_dir (str | pathlib.Path):
+            Path to the data directory to store outputs. Defaults to None, in
+            which case a subdirectory is made in the current working directory
+            with a name set by the name arg of this class.
         """
         self.name = name
-        self.simulation_class = simulation_class
-
-        # Get location of each param in the moose input file (HIT format).
-        self.hit_blocks = {}
-        self.hit_names = {}
-        for param in search_space:
-            self.hit_blocks[param["name"]] = param["hit_block"]
-            param.pop("hit_block")
-            self.hit_names[param["name"]] = param["hit_name"]
-            param.pop("hit_name")
+
+        # Add tune report step to output of design evaluation function.
```

Comment:
Adding this here with a lambda function allows the user to define their objective function with a more standard return of a dictionary.